test/extended/util/cli.go

```diff
@@ -85,21 +85,27 @@ func NewCLI(project, adminConfigPath string) *CLI {
 
 	g.BeforeEach(client.SetupProject)
 
-	ProwGCPSetup(client)
-
 	return client
 }
 
 // ProwGCPSetup makes sure certain required env vars are available in the case
 // that extended tests are invoked directly via calls to ginkgo/extended.test
 func ProwGCPSetup(oc *CLI) {
 	tn := os.Getenv("OS_TEST_NAMESPACE")
+	ad := os.Getenv("ARTIFACT_DIR")
+	btd := os.Getenv("BASETMPDIR")
+	e2e.Logf("OS_TEST_NAMESPACE env setting %s, ARTIFACT_DIR env setting %s BASETMPDIR %s", tn, ad, btd)
 	if len(strings.TrimSpace(tn)) == 0 {
 		os.Setenv("OS_TEST_NAMESPACE", oc.Namespace())
+		e2e.Logf("OS_TEST_NAMESPACE env setting now %s", os.Getenv("OS_TEST_NAMESPACE"))
 	}
-	ad := os.Getenv("ARTIFACT_DIR")
 	if len(strings.TrimSpace(ad)) == 0 {
 		os.Setenv("ARTIFACT_DIR", "/tmp/artifacts")
```

Comment:
This is not threadsafe FYI

Reply:
ah ... you mean the setenv / glibc related posts I found after some searching, right? ... interesting / thanks

searching the code base some, I see a sprinkling of `os.Setenv` usage (predominantly in tests) and no utilities best as I can tell to protect it with a Mutex, etc.

I can see about a utility / replacement of existing usage in a separate PR.